runtime/autoload/man.vim

```diff
@@ -57,30 +53,70 @@ function! man#open_page(count, count1, mods, ...) abort
     noautocmd execute 'silent' a:mods 'split' fnameescape(bufname)
   endif
   let b:man_sect = sect
-  call s:read_page(path)
+  call s:put_page(page)
 endfunction
 
 function! man#read_page(ref) abort
   try
     let [sect, name] = man#extract_sect_and_name_ref(a:ref)
     let [b:man_sect, name, path] = s:verify_exists(sect, name)
+    let page = s:get_page(path)
   catch
     " call to s:error() is unnecessary
     return
   endtry
-  call s:read_page(path)
+  call s:put_page(page)
+endfunction
+
+" Handler for s:system() function.
+function! s:system_handler(jobid, data, event) abort
+  if a:event == 'stdout'
+    let self.stdout .= join(a:data, "\n")
+  elseif a:event == 'stderr'
+    let self.stderr .= join(a:data, "\n")
+  else
+    let s:shell_error = a:data
+  endif
 endfunction
 
-function! s:read_page(path) abort
+" Run a system command and timeout after 30 seconds.
+function! s:system(cmd, ...) abort
+  let opts = {
+        \ 'stdout': '',
+        \ 'stderr': '',
+        \ 'on_stdout': function('s:system_handler'),
+        \ 'on_stderr': function('s:system_handler'),
+        \ 'on_exit': function('s:system_handler'),
+        \ }
+  let jobid = jobstart(a:cmd, opts)
+
+  if jobid < 1
+    throw printf('command error %d: %s', jobid, join(a:cmd))
+  endif
+
+  let res = jobwait([jobid], 30000)
+  if res[0] == -1
+    silent! call jobstop(jobid)
+    throw printf('command timed out: %s', join(a:cmd))
+  elseif s:shell_error != 0
+    throw printf("command error (%d) %s: %s", jobid, join(a:cmd), opts.stderr)
+  endif
+
+  return opts.stdout
+endfunction
+
+function! s:get_page(path) abort
+  return s:system(['env', 'MANPAGER=cat', (empty($MANWIDTH) ? ' MANWIDTH='.winwidth(0) : ''), 'man', a:path])
```

Comment:
There is no need for preceding space in `' MANWIDTH='`, it can be just `'MANWIDTH='`